zilliqa/src/consensus.rs

```diff
@@ -2838,4 +2899,58 @@ impl Consensus {
     ) -> Result<()> {
         self.block_store.report_outgoing_message_failure(failure)
     }
+
+    pub fn tick(&mut self) -> Result<()> {
+        trace!("consensus::tick()");
+        trace!("request_missing_blocks from timer");
+
+        // Drives the block fetching state machine - see docs/fetching_blocks.md
+        if self.block_store.request_missing_blocks()? {
+            // We're syncing..
+            // Is it likely that the next thing in the buffer could be the next block?
+            let likely_blocks = self.block_store.next_proposals_if_likely()?;
+            if likely_blocks.is_empty() {
+                trace!("no blocks buffered");
+                // If there are no next blocks buffered, someone may well have lied to us about
+                // where the gaps in the view range are. This should be a rare occurrence, so in
+                // lieu of timing it out, just zap the view range gap and we'll take the hit on
+                // any rerequests.
+                self.block_store.delete_empty_view_range_cache();
+            } else {
+                likely_blocks.into_iter().for_each(|(from, block)| {
+                    trace!(
+                        "buffer may contain the next block - {0:?} v={1} n={2}",
+                        block.hash(),
+                        block.view(),
+                        block.number()
+                    );
+                    // Ignore errors here - just carry on and wait for re-request to clean up.
```

Comment:
There's another ignored `Err` here that I would argue we should propagate up for the same reason as previously.

Happy to do that in a separate PR though - As I said, the node is in the process of  to explode if this would have return an `Err` anyhow

Reply:
@JamesHinshelwood  - sounds good to me; since I'm in the area, would you prefer:

 1. Try to send all the messages and return the first error (if there is one), or
 2. Send messages until  you hit an error, then return it.
 
 ?